src/backend/distributed/operations/delete_protocol.c

```diff
@@ -224,13 +224,30 @@ master_drop_all_shards(PG_FUNCTION_ARGS)
 
 	CheckCitusVersion(ERROR);
 
+	if (!EnableDDLPropagation)
+	{
+		PG_RETURN_INT32(-1);
+	}
+
+	int returnValue = MasterDropAllShards(relationId, schemaName, relationName);
+
+	PG_RETURN_INT32(returnValue);
+}
+
+
+/*
+ * MasterDropAllShards attempts to drop all shards for a given relation.
+ */
+int
+MasterDropAllShards(Oid relationId, char *schemaName, char *relationName)
```

Comment:
Maybe just DropAllShards. Maybe move the CheckTableSchemaNameForDrop into master_drop_all_shards, since it's very specific to calling that UDF from the SQL_DROP handler.